internal/calculator/calculator.go

```diff
@@ -0,0 +1,26 @@
+// Package calculator has code, which describe
+// volume calculator.
+package calculator
+
+import (
+	"github.com/shopspring/decimal"
+	"volume_calculator/internal/builder"
+)
+
+type Calculator struct {
+}
+
+// Calculate is the implementation of counting the volume of all shapes.
+func (calc Calculator) Calculate(shape string, radius, length, width, hieght decimal.Decimal) decimal.Decimal {
+	var volume decimal.Decimal
+	builder := builder.Builder{}
+	switch shape {
+	case "sphere":
+		volume = builder.BuildSphere(radius).CalculateVolume()
+	case "pyramid":
+		volume = builder.BuildPyramid(length, width, hieght).CalculateVolume()
+	case "cylinder":
+		volume = builder.BuildCylinder(hieght, radius).CalculateVolume()
+	}
```

Comment:
I'd propose to create New() function inside the `builder` package which will receive all args from `Calculate()`.  

And on the next row you could call `shape := builder.BuildShape()`

Reply:
Also move `switch` inside the `BuildShape()` function and in `case`'s you could just call `shape.NewSphere(...)`  and so on